File: 23_climbing_Stairs/climb.py
# Climbing Stairs - Interview Question -
# Asked By Apple and Adobe
# in Python


def climbingStairs(n):
    r = [1, 1, 2]  # Initial Result Set
    if n <= 2: return r[n]

    for step in range(3, n + 1):
        r.append(r[step - 1] + r[step - 2])
    return r[n]


n = 26  # Steps
ways = climbingStairs(n)
print(ways)  # 196418

File: 23_climbing_Stairs/climb.js

// Climbing Stairs - Interview Question -
// Asked By Apple and Adobe
// in JavaScript

function climbingStairs(n) {
    let r = [1, 1, 2] // Initial Result Set
    if (n <= 2) return r[n];

    for (let step = 3; step < n + 1; step++) {
        r.push(r[step - 1] + r[step - 2])
    }
    return r[n]
}

let n = 26; // Steps
let ways = climbingStairs(n);
console.log(ways); // 196418
